mypy/test/test_source_finder.py

```diff
@@ -0,0 +1,145 @@
+from pathlib import Path
+from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple
+
+import pytest  # type: ignore  # no pytest in typeshed
+from py._path.local import LocalPath  # type: ignore  # no py in typeshed
+
+from mypy.build import BuildSource
+from mypy.find_sources import create_source_list
+from mypy.options import Options
+
+MergeFinder = Callable[
+    [Dict[str, Any], bool, Optional[Sequence[str]]], Tuple[Path, List[BuildSource]]
+]
+
+
+def create_files(folder: Path, content: Dict[str, Any]) -> None:
+    if not folder.exists():
+        folder.mkdir()
+    for key, value in content.items():
+        if key.endswith(".py") or key.endswith(".pyi"):
+            with open(str(folder / key), "wt") as file_handler:
+                file_handler.write(value)
+        elif isinstance(value, dict):
+            create_files(folder / key, value)
+
+
+@pytest.fixture()
+def merge_finder(  # type: ignore  # cannot follow this import
+    tmpdir: LocalPath, monkeypatch: Any
+) -> MergeFinder:
+    def _checker(
+        content: Dict[str, Any], merge: bool, args: Optional[Sequence[str]] = None
+    ) -> Tuple[Path, List[BuildSource]]:
+        monkeypatch.chdir(tmpdir)
+        options = Options()
+        options.merge_stub_into_src = merge
+        test_dir = str(tmpdir)
+        create_files(Path(test_dir), content)
+        targets = create_source_list(files=[test_dir] if args is None else args, options=options)
+        return Path(str(tmpdir)), targets
+
+    return _checker
+
+
+def test_source_finder_merge(merge_finder: MergeFinder) -> None:
```

Comment:
Can you put these tests inside a test class derived from `unittest.TestCase`, for consistency?

Reply:
yes, will do 👍